mantidimaging/core/operations/overlap_correction/overlap_correction.py

```diff
@@ -0,0 +1,133 @@
+
+import numpy
+import numpy as np
+
+from mantidimaging.core.operations.base_filter import BaseFilter
+from mantidimaging.core.parallel import utility as pu
+from mantidimaging.core.utility.progress_reporting import Progress
+
+if TYPE_CHECKING:
+    from mantidimaging.core.data import ImageStack
+    from mantidimaging.gui.mvp_base import BaseMainWindowView
+    from PyQt5.QtWidgets import QFormLayout, QWidget
+
+
+class OverlapCorrection(BaseFilter):
+    """
+    Overlap correction is a post-experimental data correction performed when the input fluxes are periodic,
+    accurately restoring the input flux up to a very large fraction of pixels (up to ~90%) occupied by the end of the
+    acquisition shutter which leads to large distortions of measured timing characteristics of the input flux.
+
+    Intended to be used on: Projections
+
+    When: As a pre-processing step, to correct the observed spectrum and clarify its characteristics.
+    """
+    filter_name = "Overlap Correction"
+    link_histograms = True
+    allow_for_180_projection = False
+
+    @staticmethod
+    def filter_func(images: ImageStack, progress=None) -> ImageStack:
+        """
+        :return: The ImageStack object which has been corrected using the supplied shutter information.
+        """
+        if images.shutter_count_file:
+            shutters_dir = images.shutter_count_file.source_file.parent
+            shutters = OverlapCorrection.get_shutters(shutters_dir)
+        else:
+            raise RuntimeError("No loaded Shutter Count file for this stack.")
+
+        output = pu.create_array(images.data.shape, images.dtype)
+        output.array = execute_single(images.data, shutters, progress)
+        images.shared_array = output
+        return images
+
+    @staticmethod
+    def _compute_overlap_correction(i: int, arrays: list[np.ndarray], params: dict[str, np.ndarray]):
+        array = arrays[0]
+        output = arrays[1]
+        for sb in params['shutter_breaks']:
+            if i == sb[0] or i == sb[1]:
+                output[i] = array[i]
+            elif sb[0] < i < sb[1]:
+                ss = sb[0]
+                counts = sb[2]
+                prob_occupied = numpy.sum(array[ss:i], axis=0) / counts
+                output[i] = array[i] / (1 - prob_occupied)
+
+    @staticmethod
+    def register_gui(form: QFormLayout, on_change: Callable, view: BaseMainWindowView) -> dict[str, QWidget]:
+        return {}
+
+    @staticmethod
+    def execute_wrapper(*args) -> partial:
+        return partial(OverlapCorrection.filter_func)
+
+    @staticmethod
+    def validate_execute_kwargs(kwargs: dict[str, Any]) -> bool:
+        return True
+
+    @staticmethod
+    def get_shutters(data_dir: Path) -> list[ShutterInfo]:
+        shuter_count_file = sorted(data_dir.glob("*ShutterCount.txt"))[0]
+        shutter_times_file = sorted(data_dir.glob("*ShutterTimes.txt"))[0]
+        spectra_file = sorted(data_dir.glob("*Spectra.txt"))[0]
```

Comment:
We should log these file names at debug level, so that if there are ever issues we can see which files are being used.